test/util_test.js

```diff
@@ -39,86 +39,116 @@ describe('makeCSV()', function() {
 });
 
 describe('retrieveIssues', () => {
-  let util, axiosMock;
+  let util, octoMock;
 
   beforeEach(() => {
-    axiosMock = {get: sinon.stub()};
-    util = proxyquire('../src/util.js', {axios: axiosMock});
-  });
+    const issueData = {
+      data: [
+        {
+          title: 'issue',
+          body: 'details',
+          labels: [{name: 'type: bug'}],
+        },
+        {
+          title: 'another issue',
+          body: 'more details',
+          labels: [{name: 'bug'}],
+        },
+        {
+          title: 'issue',
+          body: 'details',
+          labels: [{name: 'other'}],
+        },
+      ],
+    };
 
-  it('should pass new issue object to makeCSV', async () => {
-    let issues = [
-      {
-        title: 'issue',
-        body: 'details',
-        labels: [{name: 'type: bug'}],
-      },
-    ];
+    const getNext = sinon.stub().returns(issueData);
+    const hasNext = sinon.stub();
+    hasNext.returns(true);
+    hasNext.onCall(1).returns(false);
 
-    const path = __dirname + '/' + 'test.csv';
-    const expectedResponse = Promise.resolve({data: issues});
-    axiosMock.get.returns(expectedResponse);
+    const mockGet = sinon.stub().returns(Promise.resolve(issueData));
 
-    const result = await util.retrieveIssues('test/test_repos.txt', path);
+    octoMock = {
+      authenticate: sinon.stub(),
+      issues: {getForRepo: mockGet},
+      hasNextPage: hasNext,
+      getNextPage: getNext,
+    };
+    util = proxyquire('../src/util.js', {
+      '@octokit/rest': () => octoMock,
+    });
+  });
 
-    // update issues to match updated labels value
-    issues[0].labels = ['type: bug'];
+  it('should pass new issue object to makeCSV', async () => {
+    const label = 'type: bug';
+    const alt = 'bug'
+    const result = await util.retrieveIssues('test/test_repos.txt', label, alt);
 
-    assert(result.length === 1);
     assert(result[0].text === 'issue details');
```

Comment:
Can you assert that the length of `result` is `2`? The current assertions don't actually verify that the array doesn't have a third item.

Reply:
Replaced this assertion with result.length === 6 because issueData is passed twice while testing the hasNext function. Also, same number of issues are returned with either 0 or 1, rather than only returning issues with matching labels.